ush/python/pygfs/jedi/jedi.py

```diff
@@ -4,145 +4,177 @@
+        logger.debug(f"Writing JEDI YAML config to: {self.jedi_config.yaml}")
+        save_as_yaml(self.jedi_config.input_config, self.jedi_config.yaml)
+
+        # Link JEDI executable to run directory
+        logger.info(f"Linking JEDI executable {self.jedi_config.exe_src} to {self.jedi_config.exe}")
+        self.link_exe()
 
     @logit(logger)
-    def execute(self, task_config: AttrDict, aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
+    def execute(self) -> None:
         """Execute JEDI application
 
         Parameters
         ----------
-        task_config: AttrDict
-            Attribute-dictionary of all configuration variables associated with a GDAS task.
-        aprun_cmd: str
-            String comprising the run command for the JEDI executable.
-        jedi_args (optional): List
-            List of strings comprising optional input arguments for the JEDI executable.
+        None
 
         Returns
         ----------
-        jedi_config: AttrDict
-            Attribute-dictionary of JEDI configuration rendered from a template.
+        None
         """
 
-        chdir(task_config.DATA)
+        chdir(self.jedi_config.rundir)
 
-        exec_cmd = Executable(aprun_cmd)
-        exec_cmd.add_default_arg(self.exe)
-        if jedi_args:
-            for arg in jedi_args:
+        exec_cmd = Executable(self.jedi_config.mpi_cmd)
+        exec_cmd.add_default_arg(self.jedi_config.exe)
+        if self.jedi_config.jedi_args:
+            for arg in self.jedi_config.jedi_args:
                 exec_cmd.add_default_arg(arg)
-        exec_cmd.add_default_arg(self.yaml)
+        exec_cmd.add_default_arg(self.jedi_config.yaml)
 
+        logger.info(f"Executing {exec_cmd}")
         try:
             exec_cmd()
         except OSError:
             raise OSError(f"FATAL ERROR: Failed to execute {exec_cmd}")
         except Exception:
             raise WorkflowException(f"FATAL ERROR: An error occurred during execution of {exec_cmd}")
 
-    @staticmethod
     @logit(logger)
-    def link_exe(task_config: AttrDict) -> None:
+    def render_jcb(self, task_config: AttrDict, algorithm: Optional[str] = None) -> AttrDict:
+        """Compile a JEDI configuration dictionary from a template file and save to a YAML file
+
+        Parameters
+        ----------
+        task_config : AttrDict
+            Dictionary of all configuration variables associated with a GDAS task.
+        algorithm (optional) : str
+            Name of the algorithm used to generate the JEDI configuration dictionary.
+            It will override the algorithm set in the jedi_config.jcb_algo_yaml file.
+
+        Returns
+        ----------
+        jedi_input_config: AttrDict
+            Attribute-dictionary of JEDI configuration rendered from a template.
+        """
+
+        # Fill JCB base YAML template and build JCB config dictionary
+        if self.jedi_config.jcb_base_yaml:
```

Comment:
The pythonic way to check if `self.jedi_config` has an attribute `jcb_base_yaml` is to do:
```suggestion
        if hasattr(self.jedi_config, 'jcb_base_yaml'):
```

Reply:
@aerorahul It's actually checking for ```None```. The ```Jedi``` class constructor ensures that all the required keys, including ```jcb_base_yaml``` are present. See here https://github.com/DavidNew-NOAA/global-workflow/blob/227718caf71c188a0075007bab9745b1f3468ba8/ush/python/pygfs/jedi/jedi.py#L47